drivers/wifi/esp_at/esp.h

```diff
@@ -107,10 +107,10 @@ extern "C" {
 #define ESP_MODE_AP		2
 #define ESP_MODE_STA_AP		3
 
-#if defined(CONFIG_WIFI_ESP_AT_VERSION_1_7)
-#define ESP_CMD_CWMODE(mode) "AT+"_CWMODE"="STRINGIFY(_CONCAT(ESP_MODE_, mode))
-#else
+#if defined(CONFIG_WIFI_ESP_AT_VERSION_1_7) || defined(CONFIG_WIFI_ESP_AT_VERSION_2_0)
```

Comment:
Sorry, this is still the wrong way around.
You are using the new command arguments (`AT+CWMODE=<mode>,[autoconnect]`) with the old AT versions.

The diff here should only be changing this:
`#if defined(CONFIG_WIFI_ESP_AT_VERSION_1_7)`
to this:
`#if defined(CONFIG_WIFI_ESP_AT_VERSION_1_7) || defined(CONFIG_WIFI_ESP_AT_VERSION_2_0)`

Reply:
@JordanYates care to make a specific suggestion in a comment, and I can merge that in to see if we can get this into 3.1?

Reply:
Sorry about that, I forgot to invert the statements, and when tested I was with the 2.1 version flashed in my board...

Updated